tofhir-server/src/main/scala/io/tofhir/server/service/job/JobFolderRepository.scala

```diff
@@ -126,14 +126,18 @@ override def getJob(projectId: String, id: String): Future[Option[FhirMappingJob
     if (!jobDefinitions.contains(projectId) || !jobDefinitions(projectId).contains(id)) {
       throw ResourceNotFound("Mapping job does not exists.", s"A mapping job with id $id does not exists in the mapping job repository at ${FileUtils.getPath(jobRepositoryFolderPath).toAbsolutePath.toString}")
     }
-    // delete the mapping job from the repository
-    getFileForJob(projectId, jobDefinitions(projectId)(id)).map(file => {
-      file.delete()
+
+    Future {
```

Comment:
The use of a Future here is not correct. Notice that the following block is a Future:
```
getFileForJob(projectId, jobDefinitions(projectId)(id)).map(file => {
  file.delete()
})
```
And you wrap it within another Future.

I suppose you are trying to do something like this:
```
    // delete the mapping job from the repository
    getFileForJob(projectId, jobDefinitions(projectId)(id)).map(file => {
      file.delete()
      // delete the mapping job from the map
      jobDefinitions(projectId).remove(id)
      // delete the job from the project
      projectFolderRepository.deleteJob(projectId, id)
    })
```

Please check!